homeassistant/components/eufylife_ble/sensor.py

```diff
@@ -0,0 +1,215 @@
+        """Set the suggested unit based on the unit system."""
+        if self.hass.config.units is US_CUSTOMARY_SYSTEM:
+            return UnitOfMass.POUNDS
+
+        return UnitOfMass.KILOGRAMS
+
+
+class EufyLifeWeightSensorEntity(RestoreEntity, EufyLifeSensorEntity):
+    """Representation of an EufyLife weight sensor."""
+
+    _attr_name = "Weight"
+    _attr_native_unit_of_measurement = UnitOfMass.KILOGRAMS
+    _attr_device_class = SensorDeviceClass.WEIGHT
+
+    _weight_kg: float | None = None
+
+    def __init__(self, data: EufyLifeData) -> None:
+        """Initialize the weight sensor entity."""
+        super().__init__(data)
+        self._attr_unique_id = f"{data.address}_weight"
+
+    @property
+    def available(self) -> bool:
+        """Determine if the entity is available."""
+        return True
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the native value."""
+        return self._weight_kg
+
+    @property
+    def suggested_unit_of_measurement(self) -> str | None:
+        """Set the suggested unit based on the unit system."""
+        if self.hass.config.units is US_CUSTOMARY_SYSTEM:
+            return UnitOfMass.POUNDS
+
+        return UnitOfMass.KILOGRAMS
+
+    @callback
+    def _handle_state_update(self, *args: Any) -> None:
+        """Handle state update."""
+        state = self._data.client.state
+        if state is not None and state.final_weight_kg is not None:
+            self._weight_kg = state.final_weight_kg
+
+        super()._handle_state_update(args)
+
+    async def async_added_to_hass(self) -> None:
+        """Restore state on startup."""
+        await super().async_added_to_hass()
+
+        last_state = await self.async_get_last_state()
+        if not last_state or last_state.state in IGNORED_STATES:
+            return
+
+        last_weight = float(last_state.state)
+        last_weight_unit = last_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+
+        # Since the RestoreEntity stores the state using the displayed unit,
+        # not the native unit, we need to convert the state back to the native
+        # unit.
+        self._weight_kg = MassConverter.convert(
+            last_weight, last_weight_unit, self.native_unit_of_measurement
+        )
+
+
+class EufyLifeHeartRateSensorEntity(RestoreEntity, EufyLifeSensorEntity):
+    """Representation of an EufyLife heart rate sensor."""
+
+    _attr_name = "Heart rate"
+    _attr_icon = "mdi:heart-pulse"
+    _attr_native_unit_of_measurement = "bpm"
+
+    _heart_rate: int | None = None
+
+    def __init__(self, data: EufyLifeData) -> None:
+        """Initialize the heart rate sensor entity."""
+        super().__init__(data)
+        self._attr_unique_id = f"{data.address}_heart_rate"
+
+    @property
+    def available(self) -> bool:
+        """Determine if the entity is available."""
+        return True
+
+    @property
+    def native_value(self) -> float | None:
```

Comment:
We could use `self._attr_native_value` instead in `_handle_state_update` and `async_added_to_hass` to set the native value.